### PR TITLE
Case Insensitive sorting for groups by Group Folders association

### PR DESCRIPTION
**Pull request by ghost**
As an Admin, on the Group Folders settings, when you want to associate a folder with a group:

Currently:
The list of groups does not occur in a "correct" alphabetical order (because its case sensitive).  If you have a lot of groups, find the correct groups is really hard. For example, groups named "foo" and "Foo" are listed very distant from each other.

With this PR:
The groups are displayed in case insensitive order. In the given example "foo" and "Foo" will be displayed very very nearly.

### CODE REVIEW

**Comment by chrisfritsche** [comment]
Hi @ChristophWurst , pls, maybe can you review this small PR ? Thanks in advance!!

**Review by ChristophWurst — APPROVED** [review]
I guess this makes sense

**Comment by ChristophWurst** [comment]
Please squash your commits into a single one

**Comment by chrisfritsche** [comment]
Thx @ChristophWurst . Pls, @nickvergessen and @juliushaertl , maybe you can also make the review, in order to merge it? I you agree, of course.  Once this is merged, i will bring others suggestions. Thanks in advance!!

File: lib/private/Group/Database.php [hunk 1]
```diff
@@ -302,6 +302,9 @@ public function getGroups($search = '', $limit = null, $offset = null) {
 		}
 		$result->closeCursor();
 
+		usort($groups, function ($a, $b) {
```
Comment (nickvergessen) [inline lib/private/Group/Database.php:305]: My biggest problem is, that there is a offset+limit. So when you have 1000 groups, Foo and foo will not be fetched in the same run, so sorting differently will be even more unexpected, so I would vote against this.

Comment (nickvergessen) [reply lib/private/Group/Database.php:305]: Instead the sorting needs to be done in the UI

Comment (chrisfritsche) [reply lib/private/Group/Database.php:305]: @nickvergessen Thanks for you feedback, and you are right! When i work on this, i will request you again. Thx!!

**Comment by skjnldsv** [comment]
> @nickvergessen Thanks for you feedback, and you are right! When i work on this, i will request you again. Thx!!

As there is no feedback since a while I will close this ticket. 
If you will decide to work on this feature again and if it hasn't been fixed or implemented already, feel free to re-open and solve the various conflicts.

Thanks for the interest in Nextcloud and the effort put into this! 🙇

**Pull request closed without merging**